sql/analyzer/validation_rules.go

```diff
@@ -531,22 +558,38 @@ func validateSubqueryColumns(ctx *sql.Context, a *Analyzer, n sql.Node, scope *S
 			return true
 		}
 
-		outerScopeRowLen := len(schemas(n.Children()))
-		plan.InspectExpressionsWithNode(s.Query, func(n sql.Node, e sql.Expression) bool {
```

Comment:
This portion of the validation never actually worked, because the if condition down below as `!valid` and this block never touched `valid`.